### ISSUE TITLE
COM reference leak?

### ISSUE THREAD

**Issue by smourier** [microsoft/Windows-Camera#58]
Hi,

Hi think there's a COM reference leak here
https://github.com/microsoft/Windows-Camera/blob/master/Samples/VirtualCamera/VirtualCameraMediaSource/SimpleMediaSource.cpp#L37

    auto ptr = winrt::make_self<SimpleMediaStream>(); // +1
    m_streamList[i] = ptr.detach(); // +2

It's cause by the mix between WIL and C++/WinRT

This could be fixed by something like this for example:

    m_streamList[i].attach(ptr.detach());

**Comment by OmarBazaraa** [comment]
Correct, and the same exact leakage occurs in the following functions as well:
- `HWMediaSource::_CreateMediaStreams`
- `AugmentedMediaSource::Initialize`
- `AugmentedMediaStream::Initialize`
